src/Implementation/Application.php

```diff
@@ -7,21 +7,60 @@
 
 class Application implements ApplicationInterface
 {
+
+
+    /**
+     * @var Command []
+     */
+    private $commands = [];
+
+    /**
+     * @var array
+     */
     private $args;
+
+    /**
+     * @var string
+     */
+    private $name;
+
     /**
      * @param string $name
      */
     public function __construct($name)
     {
-        $this->args = $_SERVER['argv'];
+        $this->setName($name);
+        $this->setArgs($_SERVER['argv']);
     }
 
     /**
      * @return int exit code
      */
     public function run()
     {
-        
+
+        if(count($this->args) === 1){
+            $output = "[37m[2m[3m" . $this->name . " contains the following commands:[0m\n";
+            foreach($this->commands as $command){
+                $output .= "[37m[2m[3m - " . $command->getName() . "[0m\n";
+            }
+            $output .= "\n";
+            echo $output;
+            return 0;
+        }
+        unset($this->args[0]);
+        $this->setArgs(array_values($this->args));
+        foreach($this->commands as $command){
+            switch($command->getName() === $this->args[0]){
```

Comment:
this switch doesn't do what you intend, it will return 1 if the given file-name is not equal to the first command of the list. The test should be failing for this.